src/item.cpp

```diff
@@ -2371,6 +2376,35 @@ bool item::has_flag( const std::string &f ) const
     return ret;
 }
 
+bool item::has_property (const std::string& prop) const {
+   return type->properties.find(prop) != type->properties.end();
+}
+
+std::string item::get_property (const std::string &prop, const std::string& def) const
```

Comment:
The default return value is always required because the property *might* not exists (what should the function return in that case?). So it's not really an extraneous parameter in my opinion.